src/objects/firmware.py

```diff
@@ -1,56 +1,118 @@
+    * :attr:`device_name`,
+    * firmware :attr:`version`,
+    * firmware :attr:`release_date`, and
+    * image-contained :attr:`part`.
+
+    Additionally, each Firmware can hold user-defined tags that may be used in advanced queries to categorize and filter all firmwares present in the database.
+    It is important to understand that said tags are **separately stored** from the :attr:`objects.file.FileObject.analysis_tags`, which are propagated by analysis plugins.
     '''
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
-        #: The name of the device that the firmware runs on
+
+        #: Device name string identifier.
+        #: Can be freely defined during upload.
+        #:
+        #: This attribute is **mandatory** and shall never be `None`.
         self.device_name: str = None
-        #: The version of the firmware in no specific format
+
+        #: Firmware version string identifier.
+        #: Can be freely defined during upload.
+        #:
+        #: This attribute is **mandatory** and shall never be `None`.
         self.version: str = None
-        #: The type of device that the firmware runs on.
-        #: E.g. "router"
+
+        #: Device class string identifier.
+        #: Not all embedded appliances are the same: There are routers, IP cameras, entertainment systems, printers, and a plethora of other classes.
+        #: FACT requires a user to categorize analyzed firmware images by this attribute.
+        #: While this attribute is **mandatory**, it can be freely defined during upload.
         self.device_class: str = None
-        #: The vendor of the firmware that this object represents.
+
+        #: Device vendor string identifier.
+        #:
+        #: This attribute is **mandatory** and shall never be `None`.
         self.vendor: str = None
-        #: The part of the firmware that this object represents.
-        #: Specifies the parts of an embedded system that are contained in the firmware.
+
+        #: Specifies the parts of an embedded system that are contained in this firmware.
         #: While this meta data string can be freely defined during firmware upload,
-        #: FACT provides a preset of frequently used values: 'complete', 'kernel', ' bootloader', and 'root-fs'.
-        #: The firmware image is assumed to be 'complete' if the assigned value is an empty string.
+        #: FACT provides a preset of frequently used values: `complete`, `kernel`, `bootloader`, and `root-fs`.
+        #:
+        #: This attribute is **optional**. The firmware image is assumed to be `complete` if the assigned/default value is an empty string.
         self.part: str = ''
-        # The release date of the firmware in the format "YYYY-MM-DD".
+
+        #: Release date string of this firmware version in `ISO 8601 <https://en.wikipedia.org/wiki/ISO_8601>`_ `YYYY-MM-DD` format.
+        #:
+        #: This attribute is **optional** and may be `None`.
```

Comment:
It would be good to add the info that it is automatically set to `'1970-01-01'` if it is empty